mmengine/infer/infer.py

```diff
@@ -0,0 +1,468 @@
+        """
+        if isinstance(inputs, str) and osp.isdir(inputs):
+            inputs = [osp.join(inputs, file) for file in os.listdir(inputs)]
+
+        if not isinstance(inputs, list):
+            inputs = [inputs]
+
+        dataloader = self._get_chunk_data(
+            map(self.pipeline, inputs), batch_size)
+        yield from map(self.collate_fn, dataloader)
+
+    @torch.no_grad()
+    def forward(self, inputs: Union[dict, tuple], **kwargs) -> Any:
+        """Forward the inputs to the model."""
+        return self.model.test_step(inputs)
+
+    @abstractmethod
+    def visualize(self,
+                  inputs: list,
+                  preds: Any,
+                  show: bool = False,
+                  **kwargs) -> List[np.ndarray]:
+        """Visualize predictions.
+
+        Customize your visualize by overriding this methods. visualize should
+        return a visualization result, it could be np.ndarray or any other
+        objects
+
+        Args:
+            inputs (list): Inputs preprocessed by :meth:`preprocess`.
+            preds (Any): Predictions of the model.
+            show (bool): Whether to display the image in a popup window.
+                Defaults to False.
+        """
+        raise NotImplementedError('visualize is not implemented!')
+
+    @abstractmethod
+    def postprocess(
+        self,
+        preds: Any,
+        visualization: List[np.ndarray],
+        return_datasample=False,
+        **kwargs,
+    ) -> dict:
+        """Postprocess predictions.
+
+        Customize your postprocess by overriding this method. postprocess
+        should return a dict with visualization results and inference results.
+
+        Args:
+            preds (List[Dict]): Predictions of the model.
+            visualization (np.ndarray): Visualized predictions.
+            return_datasample (bool): Whether to return results as datasamples.
+
+        Returns:
+            dict: Inference and visualization results with key ``predictions``
+            and ``visualization``
+
+            - ``visualization``: Returned by :meth:`visualize`
+            - ``predictions``: Returned by :meth:`postprocess` and
+                processed in :meth:`postprocess`
+        """
+        raise NotImplementedError('postprocess is not implemented!')
+
+    def _load_model_from_metafile(self, model: str) -> Tuple[Config, str]:
+        """Load config and weights from metafile.
+
+        Args:
+            model (str): model name defined in metafile.
+
+        Returns:
+            Tuple[Config, str]: Loaded Config and weights path defined in
+                metafile.
+        """
+        scope = DefaultScope.get_current_instance().scope_name  # type: ignore
+        assert scope is not None, ('scope should be initialized if you want '
+                                   'to load config from metafile.')
+        project = PKG2PROJECT[scope]
+        package_path = get_installed_path(project)
```

Comment:
What will happen if project was not installed?

Reply:
If users want to get config from the metafile, they must install the corresponding repos, otherwise an error will be raised. I think it is reasonable

Reply:
What type of error will be raised here?